api/src/main/java/net/kyori/adventure/chat/SignedMessageImpl.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2022 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.chat;
+
+import java.security.SecureRandom;
+import java.time.Instant;
+import net.kyori.adventure.identity.Identity;
+import net.kyori.adventure.text.Component;
+import org.jetbrains.annotations.NotNull;
+
+// Used for system messages
+final class SignedMessageImpl implements SignedMessage {
+  static final SecureRandom RANDOM = new SecureRandom();
+
+  private final Instant instant;
+  private final long salt;
+  private final Component message;
+  private final String plain;
+
+  SignedMessageImpl(final Component message, final String plain) {
+    this.instant = Instant.now();
+    this.salt = RANDOM.nextLong();
+    this.message = message;
+    this.plain = plain;
+  }
+
+  @Override
+  public @NotNull Instant timeStamp() {
+    return this.instant;
+  }
+
+  @Override
+  public long salt() {
+    return this.salt;
+  }
+
+  @Override
+  public Signature signature() {
```

Comment:
Feels really weird to me we can have a "signed message" without a signature. I wonder if we just don't support sending system messages with a signed message instance?

Reply:
well "technically" a system signedmessage does have a signature. its just not used, because the uuid is a nil one. so the client doesnt even bother with the signature, it just sends it as a system message.

But it is important I think, because you can use chattypes with system messages this way.

Reply:
To follow up on this, the server can use a system signed chat message in certain situations (when dealing with some command arguments) and if they were ever included in an event on a specific platform, there should be a way to represent them with the same type.